src/gettext.cpp

```diff
@@ -202,7 +207,12 @@ void init_gettext(const char *path, const std::string &configured_language,
 #endif // ifndef _WIN32
 	}
 	else {
+#ifdef __ANDROID__
+		char lang[3] = {0};
 		 /* set current system default locale */
+		AConfiguration_getLanguage(porting::app_global->config, lang);
+		setenv("LANG", lang, 1);
+#endif
```

Comment:
Can't `initializePathsAndroid` set this? That would be cleaner than adding more platform-specific code in here.

Reply:
`initializePathsAndroid` is used to define the paths of minetest on Android platform.

What we do here is to get the system language and set it when the user does not have a custom language.

If `initializePathsAndroid` set this, the function will not match the function name, which will reduce the maintainability.

So I don't think `initializePathsAndroid` can do it.

Reply:
`initAndroid`, then

Reply:
These two methods are reasonable, but in order to reduce the workload, I still do not modify this code

Reply:
This seemed to be the correct place to do the `setenv` as all other platforms do the same here.
Can probably be done somewhere else, but in that case, it could be a re-factoring for all platforms no ?

Reply:
The issue is not setenv, the issue is that we have to pull in Android-specific code to retrieve the system language (which no other platform currently needs). It would be far easier to set the language unconditionally at startup to keep it contained within `porting.cpp`.

Reply:
@oleastre 
Through my test, this code does not automatically detect the language environment.

I agree with @sfan5 that write a function in porting_android.cpp to return the language environment.

And call it in gettext.cpp.

Reply:
So does the current state of this PR detect the language correctly?

Reply:
It does not work.